tendrl/node_agent/alert/utils.py

```diff
@@ -0,0 +1,122 @@
+    except (AttributeError, EtcdException) as ex:
+        raise ex
+    if hasattr(job, 'leaves'):
+        for item in job.leaves:
+            if key == item.key:
+                result[item.key.split("/")[-1]] = item.value
+                return result
+            if item.dir is True:
+                result[item.key.split("/")[-1]] = read(item.key)
+            else:
+                result[item.key.split("/")[-1]] = item.value
+    return result
+
+
+def get_alerts():
+    alerts_arr = []
+    try:
+        alerts = read('/alerting/alerts')
+    except EtcdKeyNotFound:
+        return alerts_arr
+    except (EtcdException, AttributeError) as ex:
+        raise ex
+    for alert_id, alert in alerts.iteritems():
+        alerts_arr.append(AlertUtils().to_obj(alert))
+    return alerts_arr
+
+
+def classify_alert(alert):
+    if 'alert_catagory' in alert.tags:
+        if alert.tags['alert_catagory'] == "node":
+            NodeAlert(
+                alert_id=alert.alert_id,
+                node_id=alert.node_id,
+                time_stamp=alert.time_stamp,
+                resource=alert.resource,
+                current_value=alert.current_value,
+                tags=alert.tags,
+                alert_type=alert.alert_type,
+                severity=alert.severity,
+                significance=alert.significance,
+                ackedby=alert.ackedby,
+                acked=alert.acked,
+                ack_comment=alert.ack_comment,
+                acked_at = alert.acked_at,
+                pid=alert.pid,
+                source=alert.source,
+                delivered=alert.delivered
+            ).save()
+        elif alert.tags['alert_catagory'] == "cluster":
+            ClusterAlert(
+                alert_id=alert.alert_id,
+                node_id=alert.node_id,
+                time_stamp=alert.time_stamp,
+                resource=alert.resource,
+                current_value=alert.current_value,
+                tags=alert.tags,
+                alert_type=alert.alert_type,
+                severity=alert.severity,
+                significance=alert.significance,
+                ackedby=alert.ackedby,
+                acked=alert.acked,
+                ack_comment=alert.ack_comment,
+                acked_at = alert.acked_at,
+                pid=alert.pid,
+                source=alert.source,
+                delivered=alert.delivered
+            ).save()
+
+def update_alert_count(alert, existing_alert):
+    counter_objects = []
+    if alert.tags['alert_catagory'] == "node":
+        node_counter_obj = NodeAlertCounters(
+            node_id=alert.node_id).load()
+        counter_objects.append(node_counter_obj)
+    elif alert.tags['alert_catagory'] == "cluster":
+        cluster_counter_obj = ClusterAlertCounters(
+            integration_id=alert.tags['integration_id']
+        ).load()
+        counter_objects.append(cluster_counter_obj)
+        if alert.resource == "volume_utilization_alert":
```

Comment:
Hard coding gluster terminology in node-agent is probably not good.
Also you need to acquire lock from the stage of reading to the stage of updating as you can expect different node-agents to be simultaneously trying to do this update for different volumes(I understand duplicate alerts for same resource is not a possibility due to the design of alert cycle)

Reply:
yes i feel the same but problem is i have to identify the alert is volume related alert or not so here i need something to identify.

and i have created volume counter object also in node-agent, i feel that is also not a good approach but i need volume counter object in node-agent to store volume alert count. am i move that object to common ?

Reply:
@r0h4n @brainfunked What do you suggest?

Reply:
@r0h4n need your suggestion

Reply:
@anmolbabu yes it is like repetitive, one more problem is @brainfunked told like only node-agent has to read and write alerts in  etcd. if i follow the above approach then monitoring-integration also has to read alerts from etcd and check which is cleared event or new event or existing event with the same severity.